### PR TITLE
FIX issue where the customQueryParams weren't being applied to the eject iframe button

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybooks/storybook/issues/6800

## What I did
Fix it

### CODE REVIEW

File: lib/ui/src/components/preview/preview.js [hunk 1]
```diff
@@ -187,14 +187,15 @@ const getTools = memoize(10)(
...
+  shouldComponentUpdate({ storyId, viewMode, options, queryParams }) {
     const { props } = this;
 
     return (
       options.isFullscreen !== props.options.isFullscreen ||
       options.isToolshown !== props.options.isToolshown ||
       viewMode !== props.viewMode ||
-      storyId !== props.storyId
+      storyId !== props.storyId ||
+      queryParams !== props.queryParams
```
Comment (shilman) [inline lib/ui/src/components/preview/preview.js]: Does this need to be a deep equals?

**Review by shilman — CHANGES_REQUESTED** [review]
See comment

File: lib/ui/src/components/preview/preview.js (hunk 1 above)
Comment (ndelangen) [reply lib/ui/src/components/preview/preview.js]: no, customQueryParams is a new instance every time it's changed by setQueryParams

https://github.com/storybooks/storybook/blob/next/lib/api/src/modules/url.ts#L130-L137

File: addons/knobs/src/components/Panel.js [hunk 2]
```diff
@@ -135,7 +136,18 @@ export default class KnobPanel extends PureComponent {
...
+
+      const queryParams = {};
+
+      Object.keys(newKnobs).forEach(n => {
+        const knob = newKnobs[n];
+        queryParams[`knob-${n}`] = Types[knob.type].serialize(knob.value);
+      });
+
+      api.setQueryParams(queryParams);
+    });
```
Comment (leoyli) [inline addons/knobs/src/components/Panel.js]: I did see these code before, are they duplicated?!

File: lib/ui/src/components/preview/preview.js (hunk 1 above)
Comment (leoyli) [reply lib/ui/src/components/preview/preview.js]: Oh nice, this also fixed the problem for `addon-contexts` that eject button won't update the url.

File: addons/knobs/src/components/Panel.js (hunk 2 above)
Comment (ndelangen) [reply addons/knobs/src/components/Panel.js]: Yes, it is in `Panel.setknobs` & `Panel.handleChange`, I don't want to refactor knobs right now, considering this is for the 5.1 release,

But this duplication would be nice to remove.

**Pull request merged**